### PR TITLE
Small optimizations and cleanups

### PR DESCRIPTION
**Pull request by Limero**

### CODE REVIEW

File: complete.go [hunk 1]
```diff
@@ -215,9 +215,9 @@ func matchExec(s string) (matches []string, longest []rune) {
 		}
 	}
 
-	sort.Strings(words)
-
 	if len(words) > 0 {
+		sort.Strings(words)
```
Comment (Limero) [inline complete.go:219]: No need to sort if there are no words

File: misc.go [hunk 2]
```diff
@@ -332,13 +335,15 @@ var (
 	reWordEnd = regexp.MustCompile(`(\pL|\pN)([^\pL\pN]|$)`)
 )
 
+// TODO: Remove when bumping to Go 1.21+
```
Comment (joelim-work) [inline misc.go:338]: Just leaving a note here, I think it's fine to upgrade to the latest version of Go (do this in a separate PR). I think it is a good time now since a new version has been recently released, and it will be a while until the next one,

**Review by joelim-work — APPROVED** [review]
